bin/p2-finish-env-extractor/main.go

```diff
@@ -30,7 +36,19 @@ func main() {
 		Logger:       logging.DefaultLogger,
 	}
 
-	err := envExtractor.WriteFinish(*exitCode, *exitStatus)
+	var err error
+	// WriteFinish() is writing to a sqlite database that may be shared
+	// across many pods across a physical host. The nature of sqlite is
+	// that writes lock the entire database, which can cause transient
+	// failures. To reduce the likelihood of writes fighting each other, we
+	// retry writes a few times after they fail
+	for i := 0; i < *retries; i++ {
+		err = envExtractor.WriteFinish(*exitCode, *exitStatus)
+		if podprocess.IsRetryable(err) {
```

Comment:
1. Trying to ensure that this acts corrctly when `err` nil:

https://play.golang.org/p/SzvU4OuQ0R

Check.

2. But if err is nil, shouldn't we stop retrying? As far as I can tell, this *always* retries the full number of times.